lib/admin.rb

```diff
@@ -0,0 +1,262 @@
+require 'csv'
+require 'pry'
+
+#REFACTOR_IDEAS:
+# create helper method that creates instance of time, and a range
+
+class Admin
+  attr_reader :reservations, :find_reservation, :rooms, :vacant_rooms, :booked_rooms
+  def initialize
+    @reservations = load_reservations('spec/test_data/test_reservation.csv')
+    @rooms = create_rooms(20)
+    sort_reservations
+  end
+
+  def create_rooms(number)
+    rooms = []
+    number.times do |num|
+      input_data = {}
+      room_number = num + 1
+      input_data[:number] = room_number
+      rooms << Room.new(input_data)
+    end
+    return rooms
+  end
+
+  def load_reservations(filename)
+    reservations = []
+
+    CSV.read(filename, headers: true).each do |line|
+      data = line.to_h
+      input_data = {}
+      input_data[:id] = data["id"].to_i
+      input_data[:start_time] = Time.parse(data["start_time"])
+      input_data[:end_time] = Time.parse(data["end_time"]) # no need to subtract last day because reservation class only calculates cost
+      # REFACTOR idea: make the reservation instance a separate method - isolating dependencies
+      reservations << Reservation.new(input_data)
+    end
+
+    return reservations
+  end
+
+  #As an administrator, I can reserve a room for a given date range
+  # uses view_vacant_rooms create range with one day less
+  def reserve_room(start_date, end_date)
+    start_date = Time.parse(start_date)
+    end_date = Time.parse(end_date)
+    range = create_hotel_range(start_date, end_date) # for hotel
+    e_date = end_date - 1 # rooms do not keep track of last night
+    vacant_rooms = view_vacant_rooms(start_date, e_date)
+    if vacant_rooms.nil?
```

Comment:
Shouldn't this method instead return an empty array if there are no rooms available?